Assets/Scripts/HUD/GameResources.cs

```diff
@@ -52,35 +53,66 @@ public class GameResources : MonoBehaviour {
 		
 		if (GUI.Button (new Rect (Screen.width - 170, 90, 150, 30), "Top 3 players")) {
 			showTopListDialog = true;
+            getTopPlayers();
 		}
-		
-		if (showTopListDialog) {
-			GUI.BeginGroup(new Rect(Screen.width/2 - 150, Screen.height/2 - 100, 300, 200), GUI.skin.box);
-			
-			style.font = skin.font;
-			style.fontSize = 20;
-			style.alignment = TextAnchor.UpperCenter;
-			
-			GUIExtended.Label(new Rect(20, 10, 250, 50), " Top players ranking", style, Color.black, Color.white);
-			
-			style.fontSize = 15;
-			style.alignment = TextAnchor.UpperCenter;
-			
-			getTopPlayers();
-			
-			GUIExtended.Label(new Rect(10, 50, 250, 50), topPlayerNames[0] + "\t" + topPlayerScores[0], style, Color.black, Color.white);
-			GUIExtended.Label(new Rect(10, 100, 250, 50), topPlayerNames[1] + "\t" + topPlayerScores[1], style, Color.black, Color.white);
-			GUIExtended.Label(new Rect(10, 150, 250, 50), topPlayerNames[2] + "\t" + topPlayerScores[2], style, Color.black, Color.white);
-			
-			GUI.EndGroup();
-		}
-		
-		if (Input.GetKeyDown ("return") || Input.GetMouseButtonDown (0)) {
-			showTopListDialog = false;
-		}
-	}
-	
-	private void getTopPlayers() {
+
+        if (showTopListDialog)
+        {
+            GUI.BeginGroup(new Rect(Screen.width / 2 - 150, Screen.height / 2 - 100, 300, 200), GUI.skin.box);
+
+            style.font = skin.font;
+            style.fontSize = 20;
+            style.alignment = TextAnchor.UpperCenter;
+
+            GUIExtended.Label(new Rect(20, 10, 250, 50), " Top players ranking", style, Color.black, Color.white);
+
+            style.fontSize = 15;
+            style.alignment = TextAnchor.UpperCenter;
+
+            //getTopPlayers();
+
+            GUIExtended.Label(new Rect(10, 50, 250, 50), topPlayerNames[0] + "\t" + topPlayerScores[0], style, Color.black, Color.white);
+            GUIExtended.Label(new Rect(10, 100, 250, 50), topPlayerNames[1] + "\t" + topPlayerScores[1], style, Color.black, Color.white);
+            GUIExtended.Label(new Rect(10, 150, 250, 50), topPlayerNames[2] + "\t" + topPlayerScores[2], style, Color.black, Color.white);
+
+            GUI.EndGroup();
+        }
+
+        if (Input.GetKeyDown("return") || Input.GetMouseButtonDown(0))
+        {
+            showTopListDialog = false;
+
+            //// Test for TopPlayers script from different script
+            //// pass function that uses data as shown below
+
+            //GetComponent<TopPlayers>().requestTopPlayers(testFunc);
+
+            //// Test currentlyOnline from different script
+            //GetComponent<CurrentlyOnline>().requestOnlinePlayers(testFunc2);
+        }
+    }
+
+    ////// Write gui logic for using top player data and pass to the getTopPlayers Method as above
```

Comment:
I think we can just remove code instead of commenting it out if we will never use it. It will be in Git history forever regardless.